pkg/logql/log/parser.go

```diff
@@ -200,12 +201,16 @@ func unescapeJSONString(b []byte) string {
 		return ""
 	}
 	res := string(bU)
-	// rune error is rejected by Prometheus
-	for _, r := range res {
+
+	// rune error is rejected by Prometheus hence replacing them with space
+	removeInvalidUtf := func(r rune) rune {
 		if r == utf8.RuneError {
-			return ""
+			return -1
 		}
+		return r
 	}
+	res = strings.Map(removeInvalidUtf, res)
```

Comment:
see:  https://github.com/grafana/loki/pull/14068#issuecomment-2337504002
```suggestion
	if strings.ContainsRune(source, utf8.RuneError) {
		res = strings.Map(removeInvalidUtf, res)
	}
```